community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/LazyInstanceSelector.java

```diff
@@ -19,51 +19,64 @@
  */
 package org.neo4j.kernel.impl.index.schema.fusion;
 
-import java.util.function.IntFunction;
+import java.util.EnumMap;
+import java.util.function.Function;
 
 /**
  * Selects an instance given a certain slot and instantiate them lazy.
  * @param <T> type of instance
  */
 class LazyInstanceSelector<T> extends InstanceSelector<T>
 {
-    private final IntFunction<T> factory;
+    private final Function<IndexSlot,T> factory;
 
     /**
-     * @param instances uninstantiated instances, instantiated lazily by the {@code factory}
-     * @param factory {@link IntFunction} for instantiating instances for specific slots.
+     * Empty lazy selector
+     *
+     * See {@link this#LazyInstanceSelector(EnumMap, Function)}
+     */
+    LazyInstanceSelector( Function<IndexSlot,T> factory )
+    {
+        this( new EnumMap<>( IndexSlot.class ), factory );
+    }
+
+    /**
+     * Lazy selector with initial mapping
+     *
+     * @param map with initial mapping
+     * @param factory {@link Function} for instantiating instances for specific slots.
      */
-    LazyInstanceSelector( T[] instances, IntFunction<T> factory )
+    LazyInstanceSelector( EnumMap<IndexSlot,T> map, Function<IndexSlot,T> factory )
     {
-        super( instances );
+        super( map );
         this.factory = factory;
     }
 
     /**
      * Instantiating an instance if it hasn't already been instantiated.
      *
-     * See {@link InstanceSelector#select(int)}
+     * See {@link InstanceSelector#select(IndexSlot)}
      */
     @Override
-    T select( int slot )
+    T select( IndexSlot slot )
     {
-        if ( instances[slot] == null )
+        if ( !instances.containsKey( slot ) )
```

Comment:
computeIfAbsent?

Reply:
We need to `assertOpen()` though.

Reply:
we can do as part of 'compute' step right?